### ISSUE TITLE
Implement support for "browser" platform

### ISSUE THREAD

**Issue by bozdoz** [silkimen/cordova-plugin-advanced-http#11]
Kind of a bummer to have to have an if statement to check if it's being viewed via browser platform.

**Comment by silkimen** [comment]
Yes, you're right. It would be nice to have same functionality for browser platform. But it won't be possible to support all functions. I'll need to figure out which ones can be implemented for browser and which ones not.
But how do we handle functions which won't be supported? Throwing an error? You'll still need to have an if statement then. Do you have an idea?

**Comment by bozdoz** [comment]
I think an object with similar methods and similar responses, and with unimplemented methods throwing errors.  You could do something simple, like:

```javascript
var ProxyHTTP = {
	_request: function (method, url, data, headers, success, fail) {
		var xhr = new XMLHttpRequest();

		// defaults 
		data = data || {};
		data = this._getParams(data);
		headers = headers || {};
		success = success || function () {};
		fail = fail || function () {};

		for (var header in headers) {
			xhr.setRequestHeader(header, headers[header]);
		}

		xhr.onload = function () {
		    if (xhr.readyState === xhr.DONE &&
		        xhr.status === 200) {
		    	success({
		    		status: xhr.status,
		    		data: (typeof(xhr.responseText) === 'string') ? xhr.responseText : xhr.response,
		    		headers: xhr.getAllResponseHeaders()
		    	});
		    }
		};

		xhr.onerror = xhr.ontimeout = function () {
			fail({
				status: xhr.status,
				message: xhr.statusText,
	    		headers: xhr.getAllResponseHeaders()
			});
		};

		xhr.open(method || 'get', url, true);
		xhr.send(data && data || null);
	},
	_getParams: function (obj) {
		var output = [];

		if (typeof(obj) === 'string') {
			return obj;
		}

		for (var key in obj) {
			output.push(encodeURIComponent(key) + 
				'=' + (obj[key] === null) ? "" : obj[key]);
		}

		return output.join('&');
	},
	get: function (url, data, headers, success, fail) {
		return this._request('get', url, data, headers, success, fail);
	},
	post: function (url, data, headers, success, fail) {
		return this._request('post', url, data, headers, success, fail);
	},
	setHeader: function (header, value) {
		this.headers = this.headers || {};
		this.headers[header] = value;
	},
	// not implemented
	getBasicAuthHeader: function () {
		console.error('This method is not implemented for the browser');
	},
	useBasicAuth: function () {
		console.error('This method is not implemented for the browser');
	},
	setDataSerializer: function () {
		console.error('This method is not implemented for the browser; but it would be a good one to implement');
	}
};
```

**Comment by bozdoz** [comment]
Guess I didn't actually implement `setHeader` properly, but you get the idea :D

**Comment by januslo** [comment]
Hi,any lucky >?

**Comment by silkimen** [comment]
I've implemented a minimum set of functions which can be implemented in the browser. Hope this will meet your requirements!

Will be included in upcoming version 1.11.0.

**Comment by silkimen** [comment]
Closing this, feel free to open a new issue, if this won't solve your problem!
